### PR TITLE
Music xml conversion feature

### LINKED ISSUE #1: Voices out of sync
Thank you for providing this script. I tested it with "Da unten im Tale - Johannes Brahms" from here: http://www2.cpdl.org/wiki/index.php/Da_unten_im_Tale,_Deutsche_Volklieder_WoO_35,5_(Johannes_Brahms).

I created the midi with MuseScore from the MusicXML file, because the midi file provided for the download does not separate the voices but is a piano version.

Unfortunatley the voices seem to be out of sync in the mp3 files. Did you face similar issues? This is the version I used: [Da unten im Tale - Brahms.mid.zip](https://github.com/jackkinsella/midi-to-part-mp3s/files/3269465/Da.unten.im.Tale.-.Brahms.mid.zip)

I had to remove the default for `--common-solo-tracks`, because the script did not allow not to have one.

I also changed the instrument to piano because the sync problem is better to hear with this instrument.

[alle.mp3.zip](https://github.com/jackkinsella/midi-to-part-mp3s/files/3269467/alle.mp3.zip)

### PR DESCRIPTION
**Pull request by oli-ver**
This fixes #1: If MusicXML is converted using music21 and not MuseScore the script works. Additionally this is a nice feature.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mido==1.2.9
```
Comment (jackkinsella) [inline requirements.txt]: Oh man I love this `requirements` file approach! So much cleaner.

File: midi-to-part-mp3s [hunk 2]
```diff
@@ -26,6 +27,34 @@ class Part:
     def mp3_filepath(self):
         return self.midi_filepath.replace(".midi", ".mp3")
 
+def check_format(file_path: str) -> str:
```
Comment (jackkinsella) [inline midi-to-part-mp3s]: Sorry if this is a dumb question (this is one of my first Python projects... I normally work in Ruby & JS):

I'm wondering if the `-> str` bit in the method signature has any _effect_ (e.g. in conjunction with a certain python complier) or is it just documentation? I don't need a full answer — just a weblink about the feature would be enough so I can read up on it myself.

Whatever it is, I like it.

File: midi-to-part-mp3s [hunk 3]
```diff
@@ -26,6 +27,34 @@ class Part:
...
+      NameError: Raised if trying to convert a format not supported by the script
+  
+  Returns:
+      str -- midi file name (original or converted midi file)
+  """
+  if file_path.endswith('.mid') or file_path.endswith('.midi'):
+    return file_path
+  elif file_path.endswith('.mxl'):
+    converted_file_path = output_directory + '/temp.midi'
+    c: music21.converter = music21.converter.Converter()
```
Comment (jackkinsella) [inline midi-to-part-mp3s]: Again, another dumb question: I'm not familiar with this syntax where a variable is placed before a colon. Can you send me a link? 

Probably related, though I don't see where this `c` variable gets consumed.

File: midi-to-part-mp3s [hunk 4]
```diff
@@ -26,6 +27,34 @@ class Part:
...
+  Raises:
+      NameError: Raised if trying to convert a format not supported by the script
+  
+  Returns:
+      str -- midi file name (original or converted midi file)
+  """
+  if file_path.endswith('.mid') or file_path.endswith('.midi'):
+    return file_path
+  elif file_path.endswith('.mxl'):
+    converted_file_path = output_directory + '/temp.midi'
```
Comment (jackkinsella) [inline midi-to-part-mp3s]: Lines 45-52 are could all be summed up as conversion from musicXML to midi, so for modularity can I ask you to split apart the conversion code into a separate function, something like `convert_musicxml_to_midi()`?

File: midi-to-part-mp3s (hunk 2 above)
Comment (jackkinsella) [reply midi-to-part-mp3s]: Relatedly, this brings up an issue of consistency. If we're going to use this approach to method signatures, it should be in _all_ the functions, something we could add in a future PR.

Comment (oli-ver) [reply midi-to-part-mp3s]: It‘s a type hint and just for documentation. It has no effect on Code execution but is a major improvement when working with IDEs like VSCode. The code completion feature will use it to show documentation and code hints. It's also used by plug-ins, i. e. when generating Python docstrings.

I concur: we should add it to everything in future PRs. 

https://docs.python.org/3/library/typing.html

File: midi-to-part-mp3s (hunk 4 above)
Comment (oli-ver) [reply midi-to-part-mp3s]: Sure, I will change that

File: midi-to-part-mp3s (hunk 3 above)
Comment (oli-ver) [reply midi-to-part-mp3s]: It's also a type hint, but you are right, I am not using the variable, probably forgot to delete the line. Should also be music21.converter.Converter: c

File: requirements.txt (hunk 1 above)
Comment (oli-ver) [reply requirements.txt]: You can create it with

pip3 freeze > requirements.txt

I am using a venv, so only packages needed by this script will be included.

File: midi-to-part-mp3s (hunk 2 above)
Comment (oli-ver) [reply midi-to-part-mp3s]: I have added some additional type hints and python docstrings for better consistency

**Comment by oli-ver** [comment]
Can you please review again?

**Pull request merged**